Applications/VpView/vpFileDataSource.cxx

```diff
@@ -128,6 +134,57 @@ void vpFileDataSource::setMonitoringEnabled(bool enable)
   }
 }
 
+bool add_mie4nitf_subdatasets (const QString &path, QStringList &list) {
+#ifdef VISGUI_USE_GDAL
+    GDALAllRegister();
+
+  GDALDataset *dataset = \
+        static_cast<GDALDataset *>(GDALOpen(path.toStdString().c_str(),
+              GA_ReadOnly));
+
+    if (!dataset)
+    {
+      qWarning().nospace()<< "GDAL could not load file." <<
+        path.toStdString().c_str();
```

Comment:
Besides that this is the locale-incorrect way to convert to `std::string` (that would be [`stdString`](https://github.com/Kitware/qtextensions/blob/873c06769a2d76e2323152efc40ad910717ce648/core/qtStlUtil.h#L39)) or `char*` (that would be [`qPrintable`](https://doc.qt.io/qt-5/qtglobal.html#qPrintable)), why are you doing this at all? Just pass the `QString`...